allure-docker-api-usage/send_results.py

```diff
@@ -1,75 +1,95 @@
 import os, requests, json, base64
+from requests import Response
 
 # This directory is where you have all your results locally, generally named as `allure-results`
 allure_results_directory = '/allure-results-example'
 # This url is where the Allure container is deployed. We are using localhost as example
 allure_server = 'http://localhost:5050'
 # Project ID according to existent projects in your Allure container - Check endpoint for project creation >> `[POST]/projects`
 project_id = 'default'
-#project_id = 'my-project-id'
+# project_id = 'my-project-id'
 
+# Connection configuration
+send_results_uri = f'{allure_server}/allure-docker-service/send-results'
+generate_report_uri = f'{allure_server}/allure-docker-service/generate-report'
+ssl_verification = True
 
 current_directory = os.path.dirname(os.path.realpath(__file__))
 results_directory = current_directory + allure_results_directory
 print('RESULTS DIRECTORY PATH: ' + results_directory)
 
-files = os.listdir(results_directory)
-
-print('FILES:')
-results = []
-for file in files:
-    result = {}
 
-    file_path = results_directory + "/" + file
-    print(file_path)
+def get_file_as_result(filename):
+    file_path = results_directory + "/" + filename
 
     if os.path.isfile(file_path):
-        try:
-            with open(file_path, "rb") as f:
-                content = f.read()
-                if content.strip():
-                    b64_content = base64.b64encode(content)
-                    result['file_name'] = file
-                    result['content_base64'] = b64_content.decode('UTF-8')
-                    results.append(result)
-                else:
-                    print('Empty File skipped: '+ file_path)
-        finally :
-            f.close()
+        with open(file_path, "rb") as f:
+            content = f.read()
+            if content.strip():
+                b64_content = base64.b64encode(content)
+                result = {
+                    'file_name': filename,
+                    'content_base64': b64_content.decode('UTF-8')
+                }
+                print(f'Successfully encoded result for {file_path}')
+                return result
+            else:
+                print('Empty File skipped: ' + file_path)
     else:
-        print('Directory skipped: '+ file_path)
+        print('Directory skipped: ' + file_path)
 
-headers = {'Content-type': 'application/json'}
-request_body = {
-    "results": results
-}
-json_request_body = json.dumps(request_body)
 
-ssl_verification = True
+def gather_result_files():
+    files = os.listdir(results_directory)
+    print('FILES:')
+    results = [get_file_as_result(file) for file in files if get_file_as_result(file)]
+    return results
+
+
+def log_response(response: Response, with_report_url=False):
+    print("STATUS CODE:")
+    print(response.status_code)
```

Comment:
It will be interesting to use the logging library instead of the print method

Reply:
yes, of course. As far As I rememeber I only reworked existing script structure, did not change the logic since I have no idea how it's being used roght now